src/core/ldap/domain/ldap-event-handler.ts

```diff
@@ -152,17 +153,54 @@ export class LdapEventHandler {
             `Received PersonenkontextUpdatedEvent, personId:${event.person.id}, new personenkontexte: ${event.newKontexte.length}, deleted personenkontexte: ${event.removedKontexte.length}`,
         );
 
+        if (!event.containsAnyCurrentPKWithRollenartLehr()) {
+            // Delete all removed personenkontexte if rollenart === LEHR
+            await Promise.allSettled(
+                event.removedKontexte
+                    .filter((pk: PersonenkontextEventKontextData) => pk.rolle === RollenArt.LEHR)
+                    .map(async (pk: PersonenkontextEventKontextData) => {
```

Comment:
ineffizient, weil awaits schon in der Funktion schon await verwendet wird. Auf das Resolve sollte erst in dem allSettled gewartet werden und nicht schon innerhalb von `map(..)`.
Ist Copy-Paste von weiter unten und akzeptabel, aber nicht gut.